src/plugins/ceramic-feed/server/content-types/ceramic-post/lifecycles.js

```diff
@@ -15,34 +17,11 @@ const LifecycleHooks = new class {
     return db.query('plugin::ceramic-feed.ceramic-post')
   }
 
-  get assets() {
-    const { db } = this.strapi
-
-    return db.query('plugin::upload.file')
-  }
-
-  constructor(strapi, attributes) {
-    const fields = keys(attributes).filter(field => 'cid' !== field)
+  constructor(strapi, schema) {
+    let { fields, mediaFields } = metadata.getFieldsNames(schema)
+    fields = array.remove(fields, 'cid')
 
-    this.strapi = strapi
-    this.fields = fields
-    this.mediaFields = fields.filter(field => 'media' === get(attributes, `${field}.type`))
-  }
-
-  async beforeCreate({ params }) {
-    const { data } = params
-    const { ceramic, fields, mediaFields } = this
-    const payload = pick(data, fields)
-
-    await Promise.all(mediaFields.map(async field => {
-      const filePath = await this._filePath(payload[field])
-
-      payload[field] = filePath
-    }))
-
-    const { id } = await ceramic.create(payload)
-
-    data.cid = id
+    assign(this, { strapi, fields, mediaFields })
   }
 
   async afterUpdate({ params, result }) {
```

Comment:
the logic is:
is the publishedAt date set, then update ceramic
if publishedAt date is null then you need to read old post and check if it was previously published and now it is being unpublished. If it wasnt previously published then do nothing... do not update ceramic. if was previously published need to mark as deleted/unpublished on ceramic

Reply:
Also I dont see a match for `ceramic.update(data.cid,payload)` what is CID? where is the ceramic api receiving a cid?